azurerm_application_gateway: Error: rpc error: code = Unavailable desc = transport is closing

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

Terraform v0.12.24
+ provider.azurerm v2.6.0

### Affected Resource(s)

resource "azurerm_application_gateway"

### Terraform Configuration Files

```hcl
resource "azurerm_application_gateway" "app_gateway" {
  name                = var.app_gateway_name
  resource_group_name = azurerm_resource_group.app_gateway_rg.name
  location            = var.location

  sku {
    name     = "WAF_Medium"
    tier     = "WAF"
    capacity = 2
  }

  waf_configuration {
    enabled = "true"
    firewall_mode = "Prevention"
    rule_set_type = "OWASP"
    rule_set_version = "3.0"
  }

  gateway_ip_configuration {
    name      = local.gateway_ip_configuration_name
    subnet_id = azurerm_subnet.app_gateway_subnet.id
  }

  frontend_port {
    name = local.frontend_port_name
    port = 80
  }

  frontend_ip_configuration {
    name                          = local.frontend_ip_configuration_name
    subnet_id                     = azurerm_subnet.app_gateway_subnet.id
    private_ip_address            = var.app_gateway_private_ip_address
    private_ip_address_allocation = "Static"
    public_ip_address_id          = azurerm_public_ip.public_ip.id
  }

  backend_address_pool {
    name         = local.backend_address_pool_name
    fqdns        = var.backend_pool_fqdns
    ip_addresses = var.backend_pool_ip_addresses
  }

  backend_http_settings {
    name                  = local.http_setting_name
    cookie_based_affinity = "Disabled"
    port                  = 80
    protocol              = "Http"
    request_timeout       = 30
  }

  http_listener {
    name                           = local.listener_name
    frontend_ip_configuration_name = local.frontend_ip_configuration_name
    frontend_port_name             = local.frontend_port_name
    protocol                       = "Http"
  }

  request_routing_rule {
    name                       = local.request_routing_rule_name
    rule_type                  = "Basic"
    http_listener_name         = local.listener_name
    backend_address_pool_name  = local.backend_address_pool_name
    backend_http_settings_name = local.http_setting_name
  }
}
```

### Panic Output

azurerm_application_gateway.app_gateway: Creating...

Error: rpc error: code = Unavailable desc = transport is closing

Detailed error log:

```
2020-04-16T19:15:04.6435116Z 2020-04-16T19:15:04.630Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: }
2020-04-16T19:15:04.6436183Z 2020/04/16 19:15:04 [WARN] Provider "registry.terraform.io/-/azurerm" produced an unexpected new value for azurerm_public_ip.public_ip, but we are tolerating it because it is using the legacy plugin SDK.
2020-04-16T19:15:04.6436977Z     The following problems may be the cause of any confusing errors from downstream operations:
2020-04-16T19:15:04.6437671Z       - .tags: new element "environment" has appeared
2020-04-16T19:15:04.6438787Z [0m[1mazurerm_public_ip.public_ip: Modifications complete after 1s [id=/subscriptions/[REDACTED]/resourceGroups/[REDACTED]/providers/Microsoft.Network/publicIPAddresses/[REDACTED]][0m[0m
2020-04-16T19:15:04.6665466Z 2020/04/16 19:15:04 [WARN] Provider "registry.terraform.io/-/azurerm" produced an invalid plan for azurerm_application_gateway.app_gateway, but we are tolerating it because it is using the legacy plugin SDK.
2020-04-16T19:15:04.6668225Z     The following problems may be the cause of any confusing errors from downstream operations:
2020-04-16T19:15:04.6670700Z       - .waf_configuration[0].file_upload_limit_mb: planned value cty.NumberIntVal(100) does not match config value cty.NullVal(cty.Number)
2020-04-16T19:15:04.6675113Z       - .waf_configuration[0].max_request_body_size_kb: planned value cty.NumberIntVal(128) does not match config value cty.NullVal(cty.Number)
2020-04-16T19:15:04.6676525Z       - .waf_configuration[0].request_body_check: planned value cty.True does not match config value cty.NullVal(cty.Bool)
2020-04-16T19:15:04.6677525Z       - .identity: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
2020-04-16T19:15:04.6678529Z       - .backend_http_settings[0].pick_host_name_from_backend_address: planned value cty.False does not match config value cty.NullVal(cty.Bool)
2020-04-16T19:15:04.6679513Z       - .ssl_policy: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
2020-04-16T19:15:04.6680304Z [0m[1mazurerm_application_gateway.app_gateway: Creating...[0m[0m
2020-04-16T19:15:04.6702407Z 2020/04/16 19:15:04 [DEBUG] azurerm_application_gateway.app_gateway: applying the planned Create change
2020-04-16T19:15:04.6736961Z 2020-04-16T19:15:04.673Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: [DEBUG] setting computed for "ssl_policy" from ComputedKeys
2020-04-16T19:15:04.6741799Z 2020-04-16T19:15:04.673Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: [DEBUG] setting computed for "identity" from ComputedKeys
2020-04-16T19:15:04.6751508Z 2020-04-16T19:15:04.674Z [INFO]  plugin.terraform-provider-azurerm_v2.6.0_x5: [INFO] preparing arguments for Application Gateway creation.
2020-04-16T19:15:04.6768342Z 2020-04-16T19:15:04.675Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: [DEBUG] AzureRM Request:
2020-04-16T19:15:04.6769605Z 2020-04-16T19:15:04.675Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: GET /subscriptions/[REDACTED]/resourceGroups/[REDACTED]/providers/Microsoft.Network/applicationGateways/appgateway-sit?api-version=2019-09-01 HTTP/1.1
2020-04-16T19:15:04.6770725Z 2020-04-16T19:15:04.675Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Host: management.azure.com
2020-04-16T19:15:04.6772118Z 2020-04-16T19:15:04.675Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: User-Agent: Go/go1.12.6 (amd64-linux) go-autorest/v14.0.0 Azure-SDK-For-Go/v41.2.0 network/2019-09-01 HashiCorp Terraform/0.12.23 (+https://www.terraform.io) Terraform Plugin SDK/1.6.0 terraform-provider-azurerm/2.6.0 pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2020-04-16T19:15:04.6773411Z 2020-04-16T19:15:04.675Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Ms-Correlation-Request-Id: d71524d3-f8e7-1b7b-2440-fe6fe4da3be0
2020-04-16T19:15:04.6774252Z 2020-04-16T19:15:04.676Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Accept-Encoding: gzip
2020-04-16T19:15:04.6774975Z 2020-04-16T19:15:04.676Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5:
2020-04-16T19:15:04.6775640Z 2020-04-16T19:15:04.676Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5:
2020-04-16T19:15:04.7346282Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/[REDACTED]/resourceGroups/[REDACTED]/providers/Microsoft.Network/applicationGateways/appgateway-sit?api-version=2019-09-01:
2020-04-16T19:15:04.7357800Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: HTTP/2.0 404 Not Found
2020-04-16T19:15:04.7359271Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Content-Length: 181
2020-04-16T19:15:04.7360637Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Cache-Control: no-cache
2020-04-16T19:15:04.7362125Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Content-Type: application/json; charset=utf-8
2020-04-16T19:15:04.7363654Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Date: Thu, 16 Apr 2020 19:15:04 GMT
2020-04-16T19:15:04.7365004Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Expires: -1
2020-04-16T19:15:04.7366687Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Pragma: no-cache
2020-04-16T19:15:04.7368219Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2020-04-16T19:15:04.7369796Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Content-Type-Options: nosniff
2020-04-16T19:15:04.7371421Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Ms-Correlation-Request-Id: d71524d3-f8e7-1b7b-2440-fe6fe4da3be0
2020-04-16T19:15:04.7372934Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Ms-Failure-Cause: gateway
2020-04-16T19:15:04.7374494Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Ms-Request-Id: 10f3bfe0-b873-4d81-9bc1-cb1fbd57d233
2020-04-16T19:15:04.7376304Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: X-Ms-Routing-Request-Id: EASTUS:20200416T191504Z:10f3bfe0-b873-4d81-9bc1-cb1fbd57d233
2020-04-16T19:15:04.7377768Z 2020-04-16T19:15:04.731Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5:
2020-04-16T19:15:04.7380065Z ource 'Microsoft.Network/applicationGateways/appgateway-sit' under resource group '[REDACTED]' was not found."}}
2020-04-16T19:15:04.7412824Z [31m
2020-04-16T19:15:04.7414077Z [1m[31mError: [0m[0m[1mrpc error: code = Unavailable desc = transport is closing[0m
2020-04-16T19:15:04.7415135Z
2020-04-16T19:15:04.7415593Z [0m[0m[0m
2020-04-16T19:15:05.0141198Z panic: interface conversion: interface {} is nil, not string
2020-04-16T19:15:05.0144364Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5:
2020-04-16T19:15:05.0146347Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: goroutine 103 [running]:
2020-04-16T19:15:05.0147955Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.expandApplicationGatewayBackendAddressPools(0xc0009c4770, 0x0)
2020-04-16T19:15:05.0149864Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/resource_arm_application_gateway.go:1833 +0x847
2020-04-16T19:15:05.0151849Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.resourceArmApplicationGatewayCreateUpdate(0xc0009c4770, 0x3ba9dc0, 0xc0000c86c0, 0x0, 0x0)
2020-04-16T19:15:05.0153788Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/resource_arm_application_gateway.go:1385 +0xa56
2020-04-16T19:15:05.0155728Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0003e4e00, 0xc000902230, 0xc00102fb40, 0x3ba9dc0, 0xc0000c86c0, 0xc000b25d01, 0xc000ba7be8, 0x3b84fa0)
2020-04-16T19:15:05.0158062Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:305 +0x3b4
2020-04-16T19:15:05.0161220Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000a35f80, 0xc000b29a68, 0xc000902230, 0xc00102fb40, 0xc000ba7088, 0xc00000f708, 0x3b87100)
2020-04-16T19:15:05.0163768Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x18f
2020-04-16T19:15:05.0169451Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000f190, 0x499bf40, 0xc000e8c180, 0xc0000a23c0, 0xc00000f190, 0xc000e8c180, 0xc000b5abd0)
2020-04-16T19:15:05.0173467Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x884
2020-04-16T19:15:05.0176593Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x3f48320, 0xc00000f190, 0x499bf40, 0xc000e8c180, 0xc0000a2360, 0x0, 0x499bf40, 0xc000e8c180, 0xc000fc9000, 0x156e)
2020-04-16T19:15:05.0179356Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3189 +0x23e
2020-04-16T19:15:05.0184085Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000, 0xc000a06330, 0x8107780, 0x0, 0x0, 0x0)
2020-04-16T19:15:05.0199827Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:995 +0x466
2020-04-16T19:15:05.0209985Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000, 0x0)
2020-04-16T19:15:05.0211848Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1275 +0xda6
2020-04-16T19:15:05.0213933Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000aa5b0, 0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000)
2020-04-16T19:15:05.0215968Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:710 +0x9f
2020-04-16T19:15:05.0217101Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-04-16T19:15:05.0218241Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:708 +0xa1
2020-04-16T19:15:05.0220775Z 2020-04-16T19:15:04.739Z [DEBUG] plugin: plugin process exited: path=/workspace/components/application_gateway/src/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v2.6.0_x5 pid=66 error="exit status 2"
2020-04-16T19:15:05.0222777Z 2020/04/16 19:15:04 [DEBUG] azurerm_application_gateway.app_gateway: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0224592Z 2020/04/16 19:15:04 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0225518Z 2020/04/16 19:15:04 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0226547Z 2020-04-16T19:15:05.011Z [DEBUG] plugin: plugin exited
2020-04-16T19:15:05.0226948Z panic: interface conversion: interface {} is nil, not string
2020-04-16T19:15:05.0227591Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5:
2020-04-16T19:15:05.0228352Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: goroutine 103 [running]:
2020-04-16T19:15:05.0229753Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.expandApplicationGatewayBackendAddressPools(0xc0009c4770, 0x0)
2020-04-16T19:15:05.0231739Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/resource_arm_application_gateway.go:1833 +0x847
2020-04-16T19:15:05.0233566Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.resourceArmApplicationGatewayCreateUpdate(0xc0009c4770, 0x3ba9dc0, 0xc0000c86c0, 0x0, 0x0)
2020-04-16T19:15:05.0235603Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/resource_arm_application_gateway.go:1385 +0xa56
2020-04-16T19:15:05.0237846Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0003e4e00, 0xc000902230, 0xc00102fb40, 0x3ba9dc0, 0xc0000c86c0, 0xc000b25d01, 0xc000ba7be8, 0x3b84fa0)
2020-04-16T19:15:05.0239270Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:305 +0x3b4
2020-04-16T19:15:05.0241906Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000a35f80, 0xc000b29a68, 0xc000902230, 0xc00102fb40, 0xc000ba7088, 0xc00000f708, 0x3b87100)
2020-04-16T19:15:05.0243991Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x18f
2020-04-16T19:15:05.0246820Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000f190, 0x499bf40, 0xc000e8c180, 0xc0000a23c0, 0xc00000f190, 0xc000e8c180, 0xc000b5abd0)
2020-04-16T19:15:05.0248696Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x884
2020-04-16T19:15:05.0250785Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x3f48320, 0xc00000f190, 0x499bf40, 0xc000e8c180, 0xc0000a2360, 0x0, 0x499bf40, 0xc000e8c180, 0xc000fc9000, 0x156e)
2020-04-16T19:15:05.0253128Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3189 +0x23e
2020-04-16T19:15:05.0257885Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000, 0xc000a06330, 0x8107780, 0x0, 0x0, 0x0)
2020-04-16T19:15:05.0259905Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:995 +0x466
2020-04-16T19:15:05.0261396Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000, 0x0)
2020-04-16T19:15:05.0263411Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1275 +0xda6
2020-04-16T19:15:05.0267301Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000aa5b0, 0xc0003a86e0, 0x49fc900, 0xc000978180, 0xc000120000)
2020-04-16T19:15:05.0269538Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:710 +0x9f
2020-04-16T19:15:05.0270948Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-04-16T19:15:05.0272646Z 2020-04-16T19:15:04.738Z [DEBUG] plugin.terraform-provider-azurerm_v2.6.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:708 +0xa1
2020-04-16T19:15:05.0274521Z 2020-04-16T19:15:04.739Z [DEBUG] plugin: plugin process exited: path=/workspace/components/application_gateway/src/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v2.6.0_x5 pid=66 error="exit status 2"
2020-04-16T19:15:05.0276087Z 2020/04/16 19:15:04 [DEBUG] azurerm_application_gateway.app_gateway: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0276805Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020-04-16T19:15:05.0277900Z 2020/04/16 19:15:04 [TRACE] EvalMaybeTainted: azurerm_application_gateway.app_gateway encountered an error during creation, so it is now marked as tainted
2020-04-16T19:15:05.0278788Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalWriteState
2020-04-16T19:15:05.0279531Z 2020/04/16 19:15:04 [TRACE] EvalWriteState: removing state object for azurerm_application_gateway.app_gateway
2020-04-16T19:15:05.0280026Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2020-04-16T19:15:05.0280549Z 2020/04/16 19:15:04 [TRACE] EvalApplyProvisioners: azurerm_application_gateway.app_gateway has no state, so skipping provisioners
2020-04-16T19:15:05.0281047Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020-04-16T19:15:05.0281595Z 2020/04/16 19:15:04 [TRACE] EvalMaybeTainted: azurerm_application_gateway.app_gateway encountered an error during creation, so it is now marked as tainted
2020-04-16T19:15:05.0282586Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalWriteState
2020-04-16T19:15:05.0283062Z 2020/04/16 19:15:04 [TRACE] EvalWriteState: removing state object for azurerm_application_gateway.app_gateway
2020-04-16T19:15:05.0283518Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalIf
2020-04-16T19:15:05.0284427Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalIf
2020-04-16T19:15:05.0284832Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2020-04-16T19:15:05.0285211Z 2020/04/16 19:15:04 [TRACE] <root>: eval: *terraform.EvalApplyPost
2020-04-16T19:15:05.0286208Z 2020/04/16 19:15:04 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0286895Z 2020/04/16 19:15:04 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020-04-16T19:15:05.0287457Z 2020/04/16 19:15:04 [TRACE] [walkApply] Exiting eval tree: azurerm_application_gateway.app_gateway
2020-04-16T19:15:05.0289332Z 2020/04/16 19:15:04 [TRACE] vertex "azurerm_application_gateway.app_gateway": visit complete
2020-04-16T19:15:05.0290571Z 2020/04/16 19:15:04 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020-04-16T19:15:05.0291454Z 2020/04/16 19:15:04 [TRACE] dag/walk: upstream of "provider.azurerm (close)" errored, so skipping
2020-04-16T19:15:05.0291919Z 2020/04/16 19:15:04 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-04-16T19:15:05.0292833Z 2020-04-16T19:15:05.011Z [DEBUG] plugin: plugin exited
2020-04-16T19:15:05.0293355Z
2020-04-16T19:15:05.0293472Z
2020-04-16T19:15:05.0293579Z
2020-04-16T19:15:05.0293846Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2020-04-16T19:15:05.0294067Z
2020-04-16T19:15:05.0294953Z Terraform crashed! This is always indicative of a bug within Terraform.
2020-04-16T19:15:05.0295339Z A crash log has been placed at "crash.log" relative to your current
2020-04-16T19:15:05.0295726Z working directory. It would be immensely helpful if you could please
2020-04-16T19:15:05.0296247Z report the crash with Terraform[1] so that we can fix this.
2020-04-16T19:15:05.0296753Z
2020-04-16T19:15:05.0297014Z When reporting bugs, please include your terraform version. That
2020-04-16T19:15:05.0297384Z information is available on the first line of crash.log. You can also
2020-04-16T19:15:05.0297989Z get it by running 'terraform --version' on the command line.
2020-04-16T19:15:05.0298214Z
2020-04-16T19:15:05.0298467Z SECURITY WARNING: the "crash.log" file that was created may contain
2020-04-16T19:15:05.0299491Z sensitive information that must be redacted before it is safe to share
2020-04-16T19:15:05.0299786Z on the issue tracker.
2020-04-16T19:15:05.0299943Z
2020-04-16T19:15:05.0300170Z [1]: https://github.com/hashicorp/terraform/issues
2020-04-16T19:15:05.0300371Z
2020-04-16T19:15:05.0300887Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2020-04-16T19:15:05.2367531Z
2020-04-16T19:15:05.2433608Z ##[error]Bash exited with code '1'.
2020-04-16T19:15:05.2469000Z ##[section]Finishing: auto-apply
```

### Steps to Reproduce


1. `terraform apply`

--- a/azurerm/internal/services/network/resource_arm_application_gateway.go
+++ b/azurerm/internal/services/network/resource_arm_application_gateway.go
@@ -1829,9 +1829,11 @@ func expandApplicationGatewayBackendAddressPools(d *schema.ResourceData) *[]netw
 		backendAddresses := make([]network.ApplicationGatewayBackendAddress, 0)
 
 		for _, ip := range v["fqdns"].([]interface{}) {
-			backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{
-				Fqdn: utils.String(ip.(string)),
-			})
+			if ip != nil {
+				backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{
+					Fqdn: utils.String(ip.(string)),
+				})
+			}
 		}
 		for _, ip := range v["ip_addresses"].([]interface{}) {
 			backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{

--- a/azurerm/internal/services/network/resource_arm_application_gateway.go
+++ b/azurerm/internal/services/network/resource_arm_application_gateway.go
@@ -116,7 +116,7 @@ func resourceArmApplicationGateway() *schema.Resource {
 							Optional: true,
 							MinItems: 1,
 							Elem: &schema.Schema{
-								Type: schema.TypeString,
+								Type:         schema.TypeString,
 								ValidateFunc: validation.NoZeroValues,
 							},
 						},
@@ -1830,11 +1830,9 @@ func expandApplicationGatewayBackendAddressPools(d *schema.ResourceData) *[]netw
 		backendAddresses := make([]network.ApplicationGatewayBackendAddress, 0)
 
 		for _, ip := range v["fqdns"].([]interface{}) {
-			if ip != nil {
-				backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{
-					Fqdn: utils.String(ip.(string)),
-				})
-			}
+			backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{
+				Fqdn: utils.String(ip.(string)),
+			})
 		}
 		for _, ip := range v["ip_addresses"].([]interface{}) {
 			backendAddresses = append(backendAddresses, network.ApplicationGatewayBackendAddress{

--- a/azurerm/internal/services/network/resource_arm_application_gateway.go
+++ b/azurerm/internal/services/network/resource_arm_application_gateway.go
@@ -117,6 +117,7 @@ func resourceArmApplicationGateway() *schema.Resource {
 							MinItems: 1,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
+								ValidateFunc: validation.NoZeroValues,
 							},
 						},
 

